### PR TITLE
feat: programmatic spender name in approve modals

### PR DESCRIPTION
**Pull request by gomesalexandre**
## Description

Yearn (which we have flagged off for some time already) and Idle are the only opportunities that actually use a ShapeShift DAO router.
All other opportunities actually interact directly with contracts.

This PR:

- Changes the header and body of the approve modal to render a newly-added `spenderName` props
- Renames `contractAddress` to `spenderContractAddress` to make things more clear
- Renames `routerName` translation key to `shapeshiftRouterName`, to make it clear that this isn't a translation to be used everywhere, but only one of many contract names
- Fixes occurrences of wrong spending contract address / asset to approve / provider icon being used
- Removes FOXy Withdraw component and related flow - there is never any approval needed to withdraw from FOXy

## Notice

- [x] Have you followed the guidelines in our [Contributing]('https://github.com/shapeshift/web/blob/develop/CONTRIBUTING.md) guide?
- [x] Have you checked to ensure there aren't other open [Pull Requests](https://github.com/shapeshift/web/pulls) for the same update/change?

## Pull Request Type

- [ ] :bug: Bug fix (Non-breaking Change: Fixes an issue)
- [ ] :hammer_and_wrench: Chore (Non-breaking Change: Doc updates, pkg upgrades, typos, etc..)
- [x] :nail_care: New Feature (Breaking/Non-breaking Change)

## Issue (if applicable)

N/A

## Testing

- Ensure spender names look good in all approvals (no need to actually broadcast an approval)
- While at it, sanity test the explorer links to ensure they are the right ones
- Also test that the asset shown as the one being approved is actually the one being approved. Spotted a few weirdnesses here, fixed as part of this PR

### Engineering

- Scrutinize the updated references - asset to approve, contract name, provider icon and spender contract address - are correct. Props renaming should make it easier to spot 🙏🏽 

### Operations

- ☝🏽 

## Screenshots (if applicable)

image
image
image
image
image
image

### CODE REVIEW

**Comment by gomesalexandre** [comment]
Current dependencies on/for this PR:
* develop
  * **PR #4190** <a href="https://app.graphite.dev/github/pr/shapeshift/web/4190" target="_blank">Graphite</a>  👈
    * **PR #4152** <a href="https://app.graphite.dev/github/pr/shapeshift/web/4152" target="_blank">Graphite</a> 

This comment was auto-generated by [Graphite](https://app.graphite.dev/github/pr/shapeshift/web/4190?utm_source=stack-comment).

File: src/features/defi/providers/fox-farming/components/FoxFarmingManager/Deposit/components/Approve.tsx [hunk 1]
```diff
@@ -184,8 +184,9 @@ export const Approve: React.FC<FoxFarmingApproveProps> = ({ accountId, onNext })
   return (
     <ReusableApprove
       asset={asset}
+      spenderName={foxFarmingOpportunity.provider}
       feeAsset={feeAsset}
-      icons={foxFarmingOpportunity?.icons}
+      icons={foxFarmingOpportunity.icons}
```
Comment (gomesalexandre) [inline src/features/defi/providers/fox-farming/components/FoxFarmingManager/Deposit/components/Approve.tsx:189]: Might as well

File: src/features/defi/providers/univ2/components/UniV2Manager/Deposit/components/Approve.tsx [hunk 2]
```diff
@@ -217,7 +218,7 @@ export const Approve: React.FC<UniV2ApproveProps> = ({ accountId, onNext }) => {
       learnMoreLink='https://shapeshift.zendesk.com/hc/en-us/articles/360018501700'
       onCancel={() => onNext(DefiStep.Info)}
       onConfirm={handleApprove}
-      contractAddress={asset1ContractAddress}
+      spenderContractAddress={asset1ContractAddress}
```
Comment (gomesalexandre) [inline src/features/defi/providers/univ2/components/UniV2Manager/Deposit/components/Approve.tsx:221]: Eyyy, easier to spot wrong stuff when naming's sane

File: src/features/defi/providers/foxy/components/FoxyManager/Withdraw/components/Withdraw.tsx [hunk 3]
```diff
@@ -168,41 +144,17 @@ export const Withdraw: React.FC<
...
+        })
+        onNext(DefiStep.Confirm)
+        dispatch({
+          type: FoxyWithdrawActionType.SET_LOADING,
+          payload: false,
         })
-
-        const allowance = bnOrZero(bn(_allowance).div(bn(10).pow(asset.precision)))
-
-        // Skip approval step if user allowance is greater than or equal requested deposit amount
```
Comment (gomesalexandre) [inline src/features/defi/providers/foxy/components/FoxyManager/Withdraw/components/Withdraw.tsx:180]: This was boilerplate from deposit that ended up being used at withdraw step, making useless XHRs in the process.
FOXy withdraw doesn't need an approval as opposed to withdrawing LPs where you need to approve the LP token - grep for `safeTransfer` in https://github.com/shapeshift/yieldies/blob/main/src/contracts/Staking.sol as a sanity check

File: src/features/defi/providers/fox-farming/components/FoxFarmingManager/Deposit/components/Approve.tsx (hunk 1 above)
Comment (0xdef1cafe) [reply src/features/defi/providers/fox-farming/components/FoxFarmingManager/Deposit/components/Approve.tsx:189]: was gonna ask when we'd do this

**Review by 0xdef1cafe — APPROVED** [review]

**Pull request merged**